external-builds/pytorch/pytorch_apex_repo.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+"""Checks out Rocm Apex.
+
+There is nothing that this script does which you couldn't do by hand, but because of
+the following, getting PyTorch sources ready to build with ToT TheRock built SDKs
+consists of multiple steps:
+
+* Sources must be pre-processed with HIPIFY, creating dirty git trees that are hard
+  to develop on further.
+* Both the ROCM SDK and PyTorch are moving targets that are eventually consistent.
+  We maintain patches for recent PyTorch revisions to adapt to packaging and library
+  compatibility differences until all releases are done and available.
+
+Primary usage:
+
+    ./pytorch_apex_repo.py checkout
+
+The checkout process combines the following activities:
+
+* Clones the pytorch repository into `THIS_MAIN_REPO_NAME` with a requested `--repo-hashtag`
+  tag (default to latest release).
+* Configures PyTorch submodules to be ignored for any local changes (so that
+  the result is suitable for development with local patches).
+* Applies "base" patches to the pytorch repo and any submodules (by using
+  `git am` with patches from `patches/pytorch_ref_to_patches_dir_name(<repo-hashtag>)/<repo-name>/base`).
+* Runs `hipify` to prepare sources for AMD GPU and commits the result to the
+  main repo and any modified submodules.
+* Applies "hipified" patches to the pytorch repo and any submodules (by using
+  `git am` with patches from `patches/<repo-hashtag>/<repo-name>/hipified`).
+* Records some tag information for subsequent activities.
+
+For one-shot builds and CI use, the above is sufficient. But this tool can also
+be used to develop. Any commits made to PyTorch or any of its submodules can
+be saved locally in TheRock by running `./pybuild.py save-patches`. If checked
+in, CI runs for that revision will incorporate them the same as anyone
+interactively using this tool.
```

Comment:
We should clean up these comments across each of the scripts. They are stale and reference old files (`pybuild.py`)

Fine to keep this consistent for now.

* https://github.com/ROCm/TheRock/blob/main/external-builds/pytorch/pytorch_audio_repo.py
* https://github.com/ROCm/TheRock/blob/main/external-builds/pytorch/pytorch_torch_repo.py
* https://github.com/ROCm/TheRock/blob/main/external-builds/pytorch/pytorch_vision_repo.py